editor/RubyScanner.cpp

```diff
@@ -140,7 +140,7 @@ bool Scanner::didBlockStart()
 
 bool Scanner::didBlockEnd()
 {
-    static const QRegExp regex(QLatin1String("24_"));
+    static const QRegExp regex(QLatin1String("(24|29)_"));
```

Comment:
This will cause the following test to fail:
```
    tokenize("a = if (foo)");
    QVERIFY(m_scanner->didBlockStart());
    QVERIFY(!m_scanner->didBlockEnd());
```

i.e. the code will think that a `)` ends a block. I possible solution is to have different tokens for `{}` and `[]()`.